core/metacling/src/TCling.cxx

```diff
@@ -2555,6 +2555,29 @@ Longptr_t TCling::ProcessLine(const char* line, EErrorCode* error/*=0*/)
                indent = HandleInterpreterException(GetMetaProcessorImpl(), mod_line, compRes, &result);
             }
          }
+      } else if (cling::DynamicLibraryManager::isSharedLibrary(fname.Data()) &&
+                 strncmp(sLine.Data(), ".L", 2) != 0) { // .x macro_C.so
+         if (gSystem->Load(fname) < 0) {
+            // Loading failed.
+            compRes = cling::Interpreter::kFailure;
+         } else {
+            if (arguments.Length() == 0) {
+               arguments = "()";
+            }
+            // We need to remove the extension. (macro_C.so or macro_C.dll)
+            Ssiz_t ext = fname.Last('.');
+            if (ext != kNPOS) {
+               fname.Remove(ext);
+            }
+            // We need to remove the automatically appended _ extension when compiling (macro_C from macro.C)
+            ext = fname.Last('_');
```

Comment:
That part only works for `ACLiC` generated libraries.  It would be legitimate for a user to generate their own library with a function entry point and call it `my_user_function` (with script `my_user_function.cxx` and library `my_user_function.so`). 

One possibility since at this point the library is already loaded, we could use:
```
gInterpreter->GetFunction(nullptr, fname); 
```
to check for existence before calling.

Reply:
done! good idea.